src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,7 +17,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<T> getAll(Class<? extends Machine> type);
```

Comment:
```suggestion
    List<T> getAll(Class<? extends T> type);
```
Let's use abstraction inside the interface